pkg/check/table_structure.go

```diff
@@ -327,17 +337,52 @@ func (c *ShardingTablesCheck) Check(ctx context.Context) *Result {
 		}
 	}
 
-	if hasAutoIncrementKeyTableNum > 1 {
-		r.State = StateWarning
-		r.ErrorMsg = fmt.Sprintf("(%d) tables %+v have auto-increment key, would conflict with each other to cause data corruption", hasAutoIncrementKeyTableNum, hasAutoIncrementKeyTables)
-		r.Instruction = "please set column mapping rules for them, or handle it by yourself"
-		r.Extra = AutoIncrementKeyChecking
+	return r
+}
+
+func (c *ShardingTablesCheck) checkAutoIncrementKey(instance, schema, table string, ctStmt *ast.CreateTableStmt, info *model.TableInfo, r *Result) bool {
+	autoIncrementKeys := c.findAutoIncrementKey(ctStmt, info)
+	for columnName, isBigInt := range autoIncrementKeys {
+		hasMatchedRule := false
+		if cm, ok1 := c.mapping[instance]; ok1 {
```

Comment:
now, if `autoIncrementKeys` found, but no `mapping` set, we will not output warning?

Reply:
maybe you want to move L369 ~ L375 out from `if` (L347) block?